src/DotLiquid.Tests/Tags/StandardTagTests.cs

```diff
@@ -701,5 +710,89 @@ public void TestLegacyKeyValueDrop()
             Assert.AreEqual("Green", drop.BeforeMethod("Last"));
             Assert.IsNull(drop.BeforeMethod("UnknownProperty"));
         }
+
+
+        [Test]
+        public void TestIncrement()
+        {
+            Helper.AssertTemplateResult(expected: "0", template: "{%increment port %}");
+            Helper.AssertTemplateResult(expected: "0 1", template: "{%increment port %} {%increment port%}");
+            Helper.AssertTemplateResult(
+                expected: "0 0 1 2 1",
+                template: "{%increment port %} {%increment starboard%} {%increment port %} {%increment port%} {%increment starboard %}");
+            Helper.AssertTemplateResult(expected: "1 2", template: "{%increment port %} {%increment port %}", localVariables: Hash.FromAnonymousObject(new { port = 1 }));
+        }
+
+        [Test]
+        public void TestIncrementIntBoundary()
+        {
+            Helper.AssertTemplateResult(
+                expected: "2147483648 2147483649",
+                template: "{%increment port %} {%increment port %}",
+                localVariables: Hash.FromAnonymousObject(new { port = 2147483648 }));
+        }
+
+        [Test]
+        public void TestIncrementDoesNotAlterAssignVariables()
+        {
+            // Sample from https://shopify.dev/api/liquid/tags/variable-tags#increment
+            Helper.AssertTemplateResult(
+                expected: @"0
+                    1
+                    2
+                    10",
+                template: @"{% assign my_number = 10 -%}
+                    {%- increment my_number %}
+                    {% increment my_number %}
+                    {% increment my_number %}
+                    {{ my_number }}");
+        }
+
+        [Test]
+        public void TestIncrementDetectsBadSyntax()
+        {
+            Assert.Throws<SyntaxException>(() => Template.Parse("{% increment %}"));
+        }
+
+        [Test]
+        public void TestDecrement()
+        {
+            Helper.AssertTemplateResult(expected: "9", template: "{%decrement port %}", localVariables: Hash.FromAnonymousObject(new { port = 10 }));
+            Helper.AssertTemplateResult(expected: "-1 -2", template: "{%decrement port %} {%decrement port%}");
+            Helper.AssertTemplateResult(
+                expected: "1 5 2 2 5",
+                template: "{%increment port %} {%increment starboard%} {%increment port %} {%decrement port%} {%decrement starboard %}",
+                localVariables: Hash.FromAnonymousObject(new { port = 1, starboard = 5 }));
+        }
+
+        [Test]
+        public void TestDecrementIntBoundary()

```

Comment:
To test decrement boundary, you also need to decrement a negative int 32 value which will pass into the int 64 threshold.